### PR TITLE
feat: Rewrite from SIO+ws-relay to protobuf+p2p-circuit

### PR DESCRIPTION
**Pull request by mkg20001**
This PR is a full rewrite of libp2p-websocket-star.
Major changes:
 - No more socket.io
 - p2p-circuit is handling relay

Server pr: https://github.com/libp2p/js-libp2p-websocket-star-rendezvous/pull/11

### CODE REVIEW

**Comment by mkg20001** [comment]
PR is ready for review

**Comment by mkg20001** [comment]
@libp2p/javascript-team Anybody mind reviewing this (and server pr)? Can I just merge that?

**Comment by diasdavid** [comment]
@mkg20001 I need to review this PR and make sure it follows the model we have to achieve with https://github.com/libp2p/js-libp2p/issues/130. I plan to prioritize that and do it this week.

File: PROTOCOL.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+```protobuf
+message IdentifyRequest {
+  required string nonce = 1;
+}
+
+message IdentifyResponse {
+  required string id = 1;
+  required string pubKey = 2;
+  required bytes signature = 3;
```
Comment (RangerMauve) [inline PROTOCOL.md]: Why not have all the same fields as the [identify protocol](https://github.com/libp2p/js-libp2p-identify/blob/master/src/message.js#L26)?

Comment (mkg20001) [reply PROTOCOL.md]: This is not the same identify protocol. Additionally in the libp2p-identify protocol the public keys never get exchanged.

**Comment by diasdavid** [comment]
## Status Update:

This PR is currently blocked on https://github.com/libp2p/js-libp2p/issues/159 which will bring things like DI setup to libp2p. Lot's of work happening there. 

Also, we will only be confident to move into a p2p-circuit solution once https://github.com/ipfs/interop/pull/6 is completely done.

Things that are part of this PR that are not part of the migration to p2p-circuit should happen in a separate PR, i.e Fixing memory leaks. @mkg20001 can you do that?

**Comment by mkg20001** [comment]
@diasdavid The memory leaks are mainly socket.io's fault.
I can try to split this pr up into multiple prs.

**Comment by diasdavid** [comment]
Check out https://www.npmjs.com/package/pull-ws which is the one we use for https://github.com/libp2p/js-libp2p-websockets

**Comment by mkg20001** [comment]
I think this is no longer needed

**Pull request closed without merging**